chrome/background.js

```diff
@@ -51,4 +71,27 @@ function onMessage(request, sender, sendResponse) {
     // Need to return true if we are planning to sendResponse asynchronously
     return true;
   }
+
+  if (request.action == "dismissOTP" && sender.tab.id in tabLogins) {
+    delete tabLogins[sender.tab.id];
+  }
+}
+
+function onTabUpdated(tabId, changeInfo, tab) {
+  if (changeInfo.status == "complete" && tabId in tabLogins) {
+    chrome.tabs.executeScript(
```

Comment:
could you please extract this `chrome.tabs.executeScript` call to a separate method and call it here and in another place? In `fillLoginForm` you can have tabId too, so that shouldn't be a problem.